lib/settings_reader/vault_resolver/engines/abstract.rb

```diff
@@ -50,18 +50,25 @@ def get_and_retry_connection(address)
           end
         end
 
-        def renew_and_retry_auth(address)
-          renew_and_retry_connection(address)
-        rescue Vault::HTTPError => e # if not authenticated, let's reauthenticate and try once more
-          raise unless auth_error?(e)
+        def renew_and_retry_auth(entry)
+          renew_and_retry_connection(entry)
+        rescue Vault::HTTPError => e
+          return process_renew_auth_error(entry) if auth_error?(e)
```

Comment:
If I'm reading this code right way this is what going to happen:

1. Vault failover happens
2. Lease renew triggered
3. Auth error happens
4. Auth initiated
5. We retry renew
6. Lease not found, but an exception will not be caught as it happened outside of the rescue block